_posts/2022-05-01-release-table-store-0.1.0.md

```diff
@@ -0,0 +1,110 @@
+which brings real-time big data computing. Users need to combine Flink with some kind of external storage.
+
+The message queue will be used in both source & intermediate stages in streaming pipeline, to guarantee the
+latency stay within seconds. There will also be a real-time OLAP system receiving processed data in streaming
+fashion and serving user’s ad-hoc queries.
+
+Everything works fine as long as users only care about the aggregated results. But when users start to care
+about the intermediate data, they will immediately hit a blocker: Intermediate kafka tables are not queryable.
+
+Therefore, users use multiple systems. Writing to a lake store like Apache Hudi, Apache Iceberg while writing to Queue,
+the lake store keeps historical data at a lower cost.
+
+There are two main issues with doing this:
+- High understanding bar for users: It’s also not easy for users to understand all the SQL connectors,
+  learn the capabilities and restrictions for each of those. Users may also want to play around with
+  streaming & batch unification, but don't really know how, given the connectors are most of the time different
+  in batch and streaming use cases.
+- Increasing architecture complexity: It’s hard to choose the most suited external systems when the requirements
+  include streaming pipelines, offline batch jobs, ad-hoc queries. Multiple systems will increase the operation
+  and maintenance complexity. Users at least need to coordinate between the queue system and file system of each
+  table, which is error-prone.
+
+The Flink Table Store aims to provide a unified storage abstraction:
+- Table Store provides storage of historical data while providing queue abstraction.
+- Table Store provides competitive historical storage with lake storage capability, using LSM file structure
+  to store data on DFS, providing real-time updates and queries at a lower cost.
+- Table Store coordinates between the queue storage and historical storage, providing hybrid read and write capabilities.
+- Table Store is a storage created for Flink, which satisfies all the concepts of Flink SQL and is the most
+  suitable storage abstraction for Flink.
+
+## Core Features
+
+Flink Table Store supports the following usage:
+- **Streaming Insert**: Write changelog streams, including CDC from the database and streams.
+- **Batch Insert**: Write batch data as offline warehouse, including OVERWRITE support.
+- **Batch/OLAP Query**: Read the snapshot of the storage, efficient querying of real-time data.
+- **Streaming Query**: Read the storage changes, ensure exactly-once consistency.
+
+Flink Table Store uses the following technologies to support the above user usages:
+- Hybrid Storage: Integrating Apache Kafka to achieve real-time stream computation.
+- LSM Structure: For a large amount of data updates and high performance queries.
+- Columnar File Format: Use Apache ORC to support efficient querying.
+- Lake Storage: Metadata and data on DFS and Object Store.
```

Comment:
I wonder whether it's necessary to expose the implementation details in the release blog post, especially when the table store is still in a preview status and implementations may change in the future.

OTOH, if we think it's still valuable to provide details here, I would suggest to add a "In this preview version" at the beginning of the paragraph.

Reply:
I think adding some details is okay here as this is the 1st version and users would still be curious about the implementations considering not so many guys involved in development.
However, I think the description looks a bit strange. The details of `Hybrid Storage`, `Columnar File Format` and `Lake Storage` are used to explain how we implement it. While the details of `LSM Structure` is used to explain why we need this. Maybe we can use `key-value store` to replace `LSM Structure` and then explain how we implement this.

Reply:
I'll organize the language and add the architecture picture.